Sync v2-6-stable with v2-6-test to release 2.6.0b1

ephraimbuddy opened the pull request:
Time for `2.6.0b1`!

hussein-awala committed:
Fix mapped tasks partial arguments when DAG default args are provided (#29913)

* Add a failing test to make it pass

* use partial_kwargs when they are provide and override only None values by dag default values

* update the test and check if the values are filled in the right order

* fix overriding retry_delay with default value when it is equal to 0

* add missing default value for inlets and outlets

* set partial_kwargs dict type to dict[str, Any] and remove type ignore comments

* create a dict for default values and use NotSet instead of None to support None as accepted value

* update partial typing by removing None type from some args and set NotSet for all args

* Tweak kwarg merging slightly

This should improve iteration a bit, I think.

* Fix unit tests

---------

Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>
(cherry picked from commit f01051a75e217d5f20394b8c890425915383101f)

ephraimbuddy committed:
Update version to 2.6.0

ephraimbuddy committed:
Add release notes

amoghrajesh committed:
Adding configuration to control retry parameters for k8s api client (#29809)

* Adding configuration to control retry parameters for k8s api client

* Handling review comments

* Fixing code bug

* Fixing failing tests

* Temporary commit with UT wip

* Fixing unit test

* Fixing the strict checks

* Handling review comments from Hussein

* Revert "Handling review comments from Hussein"

This reverts commit fa3bc260f7462c42620f694ee97b7f15c0b0b9c3.

* Fixing failing ut

* Reverting bad hack

* Updating logic in kube_client.py

Co-authored-by: Hussein Awala <hussein@awala.fr>

* Fixing unit tests

* Fixing unit tests

* Handling review comments from Ash

* Fix loading mock call args for python3.7

* Apply suggestions from code review

* fix static check

* add in 2.6.0

---------

Co-authored-by: Amogh <adesai@cloudera.com>
Co-authored-by: Hussein Awala <houssein.awala.96@gmail.com>
(cherry picked from commit dcffbb4aff090e6c7b6dc96a4a68b188424ae174)

eladkal commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,174 @@
+- Allow Users to disable SwaggerUI via configuration (#28354)
+- Show mapped task groups in graph (#28392)
+- Log FileTaskHandler to work with KubernetesExecutor's multi_namespace_mode (#28436)
+- Add Amazon Elastic Container Registry (ECR) Hook (#28279)
```
this is in providers release not part of core.. should be removed

ephraimbuddy replied:
Yeah, will go through the PR. I think this one was added by the tool because it made changes to spelling_wordlist

eladkal commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,174 @@
+- Add ``FileTrigger`` (#29265)
+- Emit DataDog statsd metrics with metadata tags (#28961)
+- add some statsd metrics for dataset (#28907)
+- Add a new SSM hook and use it in the System Test context builder (#28755)
```
provider only PR. should be removed.

eladkal commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,174 @@
+- add output format arg for ``cli.dags.trigger`` (#29224)
+- Make json and yaml available in templates (#28930)
+- Enable tagged metric names for existing Statsd metric publishing events | influxdb-statsd support (#29093)
+- Add HttpHookAsync for deferrable implementation (#29038)
```
provider only PR should be removed

bbovenzi committed:
Improve task & run actions ux in grid view (#30373)

* update run clear+mark, update task clear

* add mark as tasks and include list of affected tasks

* Add support for mapped tasks, add shared modal component

* Clean up styling, restore warning for past/future tg clear

(cherry picked from commit c5b685e88dd6ecf56d96ef4fefa6c409f28e2b22)

maahir22 committed:
Add command to get DAG Details via CLI (#30432)

---------

Co-authored-by: Hussein Awala <hussein@awala.fr>
Co-authored-by: Hussein Awala <houssein.awala.96@gmail.com>
(cherry picked from commit 7074167d71c93b69361d24c1121adc7419367f2a)

sean-rose committed:
When clearing task instances try to get associated DAGs from database (#29065)

* When clearing task instances try to get associated DAGs from database.

This fixes problems when recursively clearing task instances across multiple DAGs:
  * Task instances in downstream DAGs weren't having their `max_tries` property incremented, which could cause downstream external task sensors in reschedule mode to instantly time out (issue #29049).
  * Task instances in downstream DAGs could have some of their properties overridden by an unrelated task in the upstream DAG if they had the same task ID.

* Use session fixture for new `test_clear_task_instances_without_dag_param` test.

* Use session fixture for new `test_clear_task_instances_in_multiple_dags` test.

---------

Co-authored-by: eladkal <45845474+eladkal@users.noreply.github.com>
(cherry picked from commit 0d2e6dce709acebdb46288faef17d322196f29a2)

vandonr-amz committed:
preload airflow imports before dag parsing to save time (#30495)

---------

Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>
Co-authored-by: Ephraim Anierobi <splendidzigy24@gmail.com>
(cherry picked from commit 9fab11cf51411276f3e70a24a23795bf959784d9)

rohan472000 committed:
Create audit_logs.rst (#30405)

---------

Co-authored-by: Josh Fell <48934154+josh-fell@users.noreply.github.com>
(cherry picked from commit 959039837c903bab4909d9bc061438b1961a79f8)

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,271 @@
+  Default setting handles case where impersonation is needed and both users (airflow and the impersonated user)
+  have the same group set as main group. Previously the default was also other-writeable and the user might choose
+  to use the other-writeable setting if they wish by configuring ``file_task_handler_new_folder_permissions``
+  and ``file_task_handler_new_file_permissions`` in ``logging`` section.
```
```suggestion
Default setting handles case where impersonation is needed and both users (airflow and the impersonated user)
have the same group set as main group. Previously the default was also other-writeable and the user might choose
to use the other-writeable setting if they wish by configuring ``file_task_handler_new_folder_permissions``
and ``file_task_handler_new_file_permissions`` in ``logging`` section.
```

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,271 @@
+SLA callbacks no longer add files to the dag processor manager's queue (#30076)
+"""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
+  This stops SLA callbacks from keeping the dag processor manager permanently busy. It means reduced CPU,
+  and fixes issues where SLAs stop the system from seeing changes to existing dag files. Additional metrics added to help track queue state.
```
```suggestion
This stops SLA callbacks from keeping the dag processor manager permanently busy. It means reduced CPU,
and fixes issues where SLAs stop the system from seeing changes to existing dag files. Additional metrics added to help track queue state.
```

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,271 @@
+  This is potentially a breaking change for any custom trigger implementations that override the ``cleanup()``
+  method and uses synchronous code, however using synchronous operations in cleanup was technically wrong,
+  because the method was executed in the main loop of the Triggerer and it was introducing unnecessary delays
+  impacting other triggers. The change is unlikely to affect any existing trigger implementations.
```
```suggestion
This is potentially a breaking change for any custom trigger implementations that override the ``cleanup()``
method and uses synchronous code, however using synchronous operations in cleanup was technically wrong,
because the method was executed in the main loop of the Triggerer and it was introducing unnecessary delays
impacting other triggers. The change is unlikely to affect any existing trigger implementations.
```

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,271 @@
+""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
+  The gauge has never been working and its value has always been 0. Having an accurate
+  value for this metric is complex so it has been decided that removing this gauge makes
+  more sense than fixing it with no certainty of the correctness of its value.
```
```suggestion
The gauge has never been working and its value has always been 0. Having an accurate
value for this metric is complex so it has been decided that removing this gauge makes
more sense than fixing it with no certainty of the correctness of its value.
```

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,271 @@
+  ``[kubernetes] worker_pods_pending_timeout``, ``[celery] stalled_task_timeout``, and
+  ``[celery] task_adoption_timeout``. If any of these configurations are set, the longest timeout will be
+  respected. For example, if ``[celery] stalled_task_timeout`` is 1200, and ``[scheduler] task_queued_timeout``
+  is 600, Airflow will set ``[scheduler] task_queued_timeout`` to 1200.
```
```suggestion
Logic for handling tasks stuck in the queued state has been consolidated, and the all configurations
responsible for timing out stuck queued tasks have been deprecated and merged into
``[scheduler] task_queued_timeout``. The configurations that have been deprecated are
``[kubernetes] worker_pods_pending_timeout``, ``[celery] stalled_task_timeout``, and
``[celery] task_adoption_timeout``. If any of these configurations are set, the longest timeout will be
respected. For example, if ``[celery] stalled_task_timeout`` is 1200, and ``[scheduler] task_queued_timeout``
is 600, Airflow will set ``[scheduler] task_queued_timeout`` to 1200.
```

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,271 @@
+  was displayed at the top but it was not obvious whether this default configuration was overridden or not.
+  Subsequently, the non-documented endpoint ``/configuration?raw=true`` is deprecated and will be removed in
+  Airflow 3.0. The HTTP response now returns an additional ``Deprecation`` header. The ``/config`` endpoint on
+  the REST API is the standard way to fetch Airflow configuration programmatically.
```
```suggestion
The configurations view now only displays the running configuration. Previously, the default configuration
was displayed at the top but it was not obvious whether this default configuration was overridden or not.
Subsequently, the non-documented endpoint ``/configuration?raw=true`` is deprecated and will be removed in
Airflow 3.0. The HTTP response now returns an additional ``Deprecation`` header. The ``/config`` endpoint on
the REST API is the standard way to fetch Airflow configuration programmatically.
```

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,271 @@
+
+Explicit skipped states list for ExternalTaskSensor (#29933)
+""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
+  ExternalTaskSensor now has an explicit ``skipped_states`` list
```
```suggestion
ExternalTaskSensor now has an explicit ``skipped_states`` list
```

jedcunningham replied:
I feel like this can just be folded into the main list of improvements.

ephraimbuddy replied:
The title is already on the list but this seems different since it was from fragment

jedcunningham replied:
Ah, then I don't see much value in having this extra chunk then. Just in the list is enough for this change imo.

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,271 @@
+Handle OverflowError on exponential backoff in next_run_calculation (#28172)
+""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
+  Maximum retry task delay is set to be 24h (86400s) by default. You can change it globally via ``core.max_task_retry_delay``
+  parameter.
```
```suggestion
Maximum retry task delay is set to be 24h (86400s) by default. You can change it globally via ``core.max_task_retry_delay``
parameter.
```

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,271 @@
+Move Hive macros to the provider (#28538)
+"""""""""""""""""""""""""""""""""""""""""
+  The Hive Macros (``hive.max_partition``, ``hive.closest_ds_partition``) are available only when Hive Provider is
+  installed. Please install Hive Provider > 5.1.0 when using those macros.
```
```suggestion
The Hive Macros (``hive.max_partition``, ``hive.closest_ds_partition``) are available only when Hive Provider is
installed. Please install Hive Provider > 5.1.0 when using those macros.
```

pierrejeambrun commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,271 @@
+- Add a new parameter for base sensor to catch the exceptions in poke method (#30293)
+- Fix dag run conf encoding with non-JSON serializable values (#28777)
+- Restore graph expand/collapse scale (#30360)
+- Fix graph zoom scale (#30355)
```
I think this one was released in 2.5.3

pierrejeambrun commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,271 @@
+- Drop k8s 1.21 and 1.22 support (#28168)
+- Remove unused task_queue attr from k8s scheduler class (#28049)
+- Completed D400 for multiple folders (#27767)
+- Completed D400 for multiple folders (#27768)
```
```suggestion
- Completed D400 for multiple folders (#27767, #27768)
```

pierrejeambrun reviewed (commented):
Looking good overall, just a few minor suggestions

pierrejeambrun commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,271 @@
+- Make the webserver startup respect update_fab_perms (#30246)
+- Ignore error when changing log folder permissions (#30123)
+- Disable ordering DagRuns by note (#30043)
+- Fix some long known Graph View UI problems (#29971)
```
Also released in 2.5.3

pierrejeambrun commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,271 @@
+- Avoid logging sensitive information in triggerer job log (#30110)
+- Add a new parameter for base sensor to catch the exceptions in poke method (#30293)
+- Fix dag run conf encoding with non-JSON serializable values (#28777)
+- Restore graph expand/collapse scale (#30360)
```
Released in 2.5.3

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,267 @@
+
+Doc only changes
+""""""""""""""""
+- docs: use correct import path for Dataset (#30617)
```
```suggestion
- Use correct import path for Dataset (#30617)
```

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,267 @@
+- docs: use correct import path for Dataset (#30617)
+- Create ``audit_logs.rst`` (#30405)
+- Adding taskflow API example for sensors (#30344)
+- add clarification about timezone aware dags (#30467)
```
```suggestion
- Add clarification about timezone aware dags (#30467)
```

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,267 @@
+- Add tests to PythonOperator (#30362)
+- Add asgiref as a core dependency (#30527)
+- Discovery safe mode toggle comment clarification (#30459)
+- fix: upgrade moment-timezone package to fix Tehran tz (#30455)
```
```suggestion
- Upgrade moment-timezone package to fix Tehran tz (#30455)
```

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,267 @@
+- Add a new parameter for base sensor to catch the exceptions in poke method (#30293)
+- Fix dag run conf encoding with non-JSON serializable values (#28777)
+- Added fixes for Airflow to be usable on Windows Dask-Workers (#30249)
+- fix: force DAG last modified time to UTC (#30243)
```
```suggestion
- Force DAG last modified time to UTC (#30243)
```

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,267 @@
+- Use the engine provided in the session (#29804)
+- Type related import optimization for Executors (#30361)
+- Add more type hints to the code base (#30503)
+- some fixes to metrics doc (#30290)
```
I think we should consider this a docs change.

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,267 @@
+- Add command to export purged records to CSV files (#29058)
+- Add ``FileTrigger`` (#29265)
+- Emit DataDog statsd metrics with metadata tags (#28961)
+- add some statsd metrics for dataset (#28907)
```
```suggestion
- Add some statsd metrics for dataset (#28907)
```

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,267 @@
+- add some statsd metrics for dataset (#28907)
+- Add --overwrite option to ``connections import`` CLI command (#28738)
+- Add general-purpose "notifier" concept to DAGs (#28569)
+- add a new conf to wait past_deps before skipping a task (#27710)
```
```suggestion
- Add a new conf to wait past_deps before skipping a task (#27710)
```

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,267 @@
+- Update gantt chart UI to display queued state of tasks (#28686)
+- Add index on log.dttm (#28944)
+- Display only the running configuration in configurations view (#28892)
+- css, cap dropdown menu size dynamically (#28736)
```
```suggestion
- Cap dropdown menu size dynamically (#28736)
```

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,267 @@
+- Add index on log.dttm (#28944)
+- Display only the running configuration in configurations view (#28892)
+- css, cap dropdown menu size dynamically (#28736)
+- added JSON linter to connection edit / add UI for field extra. On connection edit screen, existing extra data will be displayed indented (#28583)
```
```suggestion
- Added JSON linter to connection edit / add UI for field extra. On connection edit screen, existing extra data will be displayed indented (#28583)
```

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,267 @@
+- Make the webserver startup respect update_fab_perms (#30246)
+- Ignore error when changing log folder permissions (#30123)
+- Disable ordering DagRuns by note (#30043)
+- try_number was not being passed to the get_task_log method, instead (#28817)
```
```suggestion
- Fix reading logs from finished KubernetesExecutor worker pod (#28817)
```

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,267 @@
+- Adding taskflow API example for sensors (#30344)
+- add clarification about timezone aware dags (#30467)
+- Clarity params documentation (#30345)
+- fix(metrics): fix unit for task duration metric (#30273)
```
```suggestion
- Fix unit for task duration metric (#30273)
```
